pkg/api/norman/customization/aks/handler.go

```diff
@@ -0,0 +1,197 @@
+	}
+}
+
+func (h *handler) getCloudCredential(req *http.Request, cap *Capabilities, credID string) (int, error) {
+	ns, name := ref.Parse(credID)
+	if ns == "" || name == "" {
+		logrus.Debugf("[AKS] invalid cloud credential ID %s", credID)
+		return http.StatusBadRequest, fmt.Errorf("invalid cloud credential ID %s", credID)
+	}
+
+	var accessCred client.CloudCredential //var to check access
+	if err := access.ByID(h.generateAPIContext(req), &schema.Version, client.CloudCredentialType, credID, &accessCred); err != nil {
+		if apiError, ok := err.(*httperror.APIError); ok {
+			if apiError.Code.Status == httperror.PermissionDenied.Status || apiError.Code.Status == httperror.NotFound.Status {
+				return httperror.InvalidBodyContent.Status, fmt.Errorf("cloud credential not found")
+			}
+		}
+		return httperror.NotFound.Status, err
+	}
+
+	cc, err := h.secretsLister.Get(ns, name)
+	if err != nil {
+		logrus.Debugf("[AKS] error accessing cloud credential %s", credID)
+		return httperror.InvalidBodyContent.Status, fmt.Errorf("error accessing cloud credential %s", credID)
+	}
+	cap.SubscriptionID = string(cc.Data["azurecredentialConfig-subscriptionId"])
+	cap.ClientID = string(cc.Data["azurecredentialConfig-clientId"])
+	cap.ClientSecret = string(cc.Data["azurecredentialConfig-clientSecret"])
+
+	cap.TenantID = req.URL.Query().Get("tenantId")
+	if cap.TenantID == "" {
+		logrus.Debugf("[AKS] error getting tenantId")
+		return http.StatusBadRequest, fmt.Errorf("error getting tenantId")
+	}
+
+	cap.BaseURL = req.URL.Query().Get("baseUrl")
+	if cap.BaseURL != "" {
+		cap.BaseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+	cap.AuthBaseURL = req.URL.Query().Get("authBaseUrl")
+	if cap.AuthBaseURL != "" {
+		cap.AuthBaseURL = azure.PublicCloud.ActiveDirectoryEndpoint
+	}
+	region := req.URL.Query().Get("region")
+	if region != "" {
+		cap.ResourceLocation = region
+	}
+
+	return http.StatusOK, nil
+}
+
+func (h *handler) getCredentialsFromBody(req *http.Request, cap *Capabilities) (int, error) {
+	raw, err := ioutil.ReadAll(req.Body)
+	if err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot read request body: %v", err)
+	}
+
+	if err = json.Unmarshal(raw, &cap); err != nil {
+		return http.StatusBadRequest, fmt.Errorf("cannot parse request body: %v", err)
+	}
+
+	if cap.SubscriptionID == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid subscriptionId")
+	}
+	if cap.TenantID == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid tenantId")
+	}
+	if cap.ClientID == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid clientId")
+	}
+	if cap.ClientSecret == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid clientSecret")
+	}
+	if cap.BaseURL == "" {
+		cap.BaseURL = azure.PublicCloud.ResourceManagerEndpoint
+	}
+	if cap.AuthBaseURL == "" {
+		cap.AuthBaseURL = azure.PublicCloud.ActiveDirectoryEndpoint
+	}
+	if cap.ResourceLocation == "" {
+		return http.StatusBadRequest, fmt.Errorf("invalid region")
```

Comment:
This seems to be introducing a backwards incompatible change. The old credential extract method didn't validate for region:

https://github.com/rancher/rancher/pull/32454/files#diff-0ed478dc297a500b1c2a4a0a0fcee2393bd367965c2188b3087a191e7d8c5545L31-L44

and region is only required to be set for aksVersions, not aksVirtualNetworks

https://github.com/rancher/rancher/pull/32454/files#diff-17b35341add493e82c30af8bd8281310f4f2ebcfa30ca6db11d9ad20f81c119bL175-L192

Reply:
indeed it is already checked in `listKubernetesVersions`